### PR TITLE
fix: convert npm to yarn

### PR DESCRIPTION
**Pull request by mkesavan13**
# Closes: [SPARK-567501](https://jira-eng-gpk2.cisco.com/jira/browse/SPARK-567501)

# Description
This PR addresses changes in the widgets repo to convert existing npm setup to a yarn monorepo. In this PR, we have moved the entire MeetingsWidgets (@webex/widgets) repo into a separate package while the newer widgets will have newer package with their names and tooling.

# Tests performed
Tested by joining my PMR from the sample app for this repo. 

- Another attendee joined
- Sending Screen share
- Mute/Unmute for both Video and Audio

### Waiting for Others to Join
waiting for joining

### Another participant joined
two participants

### Screen share in Widgets
screenshare at widget end

### Same Screen share at receiver's side
screenshare at receiver side

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -1,156 +1,12 @@
...
-  "repository": {
-    "type": "git",
-    "url": "https://github.com/webex/widgets.git"
-  },
-  "publishConfig": {
-    "tag": "latest",
-    "access": "public"
-  },
+  "name": "webex-widgets",
+  "version": "1.0.0",
```
Comment (Shreyas281299) [inline package.json:3]: Version can be anything since its the main package and its not gonna be published.

**Review by Shreyas281299 — APPROVED** [review]
Changes are fine. 
I just have 1 question that there is no yarnrc fie changes - we need to specify the node-linker in yarnrc. But since we only have 1 workspace right now its hard to know which node-linker to use, so maybe we can add the yarnrc changes as part of my PR.

**Review by Shreyas281299 — CHANGES_REQUESTED** [review]
I think there should be some automatically created files for yarn, also we need to add the yarnrc file with yarn version atleast.

```
checksumBehavior: update

nodeLinker: node-modules

npmRegistryServer: "https://registry.npmjs.org"

yarnPath: .yarn/releases/yarn-4.5.0.cjs

```
This is the yarnrc file that I have in react-widgets created with yarn@4.5.0. We can discuss this.

File: .circleci/config.yml [hunk 2]
```diff
@@ -106,4 +113,4 @@ workflows:
             - docs
           filters:
             branches:
-              only: master
+              only: master
```
Comment (mkesavan13) [inline .circleci/config.yml:116]: Add back the EoL empty line

File: .gitignore [hunk 3]
```diff
@@ -43,4 +43,4 @@ reports/
 !.env.default
 !package.json
 !package-lock.json
-node_modules/
+node_modules/
```
Comment (mkesavan13) [inline .gitignore:46]: Add back the EoL empty line

File: .yarnrc.yml [hunk 4]
```diff
@@ -0,0 +1,4 @@
+yarnPath: .yarn/releases/yarn-4.5.1.cjs
+checksumBehavior: update
+nodeLinker: node-modules
+npmRegistryServer: "https://registry.npmjs.org"
```
Comment (mkesavan13) [inline .yarnrc.yml:4]: Add the EoL empty line

File: package.json [hunk 5]
```diff
@@ -1,156 +1,16 @@
...
-        ]
-      ]
-    }
-  },
-  "husky": {
-    "hooks": {
-      "pre-commit": "npm run test:eslint",
-      "commit-msg": "commitlint -E HUSKY_GIT_PARAMS"
-    }
+    "node-gyp": "^10.2.0"
```
Comment (mkesavan13) [inline package.json:14]: This dev dependency is required to build the `deasync` dependency from `sdk-component-adapter`

File: package.json (hunk 1 above)
Comment (mkesavan13) [reply package.json]: Removed the version for better :-)

File: package.json [hunk 6]
```diff
@@ -1,156 +1,15 @@
...
-  "repository": {
-    "type": "git",
-    "url": "https://github.com/webex/widgets.git"
-  },
-  "publishConfig": {
-    "tag": "latest",
-    "access": "public"
-  },
+  "name": "webex-widgets",
+  "main": "index.js",
```
Comment (Shreyas281299) [inline package.json:3]: I don't see any index.js that can be the main for this package. In js-sdk also we have a non-existing file as main. We dont have to remove this, but if we find some documentation that explains this then it would be nice.

File: .circleci/config.yml [hunk 7]
```diff
@@ -24,63 +24,70 @@ jobs:
...
+            - packages/*/node_modules
 
   linter:
     executor: node-executor
     steps:
       - checkout_from_workspace
       - run:
           name: Linting
-          command: npm run test:eslint
+          command: yarn workspace @webex/widgets run test:eslint
```
Comment (Shreyas281299) [inline .circleci/config.yml:48]: Do you think we should update all these commands to run the commands for each package? Rather than a specific package.

```suggestion
          command: "yarn workspaces foreach -A run test:eslint
```

File: packages/@webex/widgets/package.json [hunk 8]
```diff
@@ -0,0 +1,156 @@
...
+    "start": "npm run demo:serve",
+    "test:e2e": "npm run demo:build && wdio wdio.conf.js",
+    "test:eslint": "eslint src/"
+  },
+  "repository": {
+    "type": "git",
+    "url": "https://github.com/webex/widgets.git"
+  },
+  "publishConfig": {
+    "tag": "latest",
```
Comment (Shreyas281299) [inline packages/@webex/widgets/package.json:25]: Are we planning to publish this onto the same tag in npm?

File: package.json (hunk 6 above)
Comment (mkesavan13) [reply package.json:3]: I think this came in as part of standard `yarn init`

I'll remove this

File: .circleci/config.yml (hunk 7 above)
Comment (mkesavan13) [reply .circleci/config.yml:48]: I wanted to keep that in the feature branch since we are merging this to the main branch

File: packages/@webex/widgets/package.json (hunk 8 above)
Comment (mkesavan13) [reply packages/@webex/widgets/package.json:25]: In this PR, this is correct so that if we do any bug fixes mid-way, we can publish it to the latest. We will change it in the feature branch later.

File: .circleci/config.yml (hunk 7 above)
Comment (Shreyas281299) [reply .circleci/config.yml:48]: Sure.

**Review by Shreyas281299 — APPROVED** [review]

**Review by adhmenon — APPROVED** [review]
LGTM

File: .yarn/install-state.gz
Comment (sreenara) [inline .yarn/install-state.gz:1]: This file shouldn't be committed to the repo

File: .yarn/releases/yarn-4.5.1.cjs
Comment (sreenara) [inline .yarn/releases/yarn-4.5.1.cjs:1]: Is this file required? I've not seen this committed to a repo

**Review by sreenara — COMMENTED** [review]
I'm unable to load all the files for review.
Additionally, there seems to be a lot of demo bundles inside the docs/ folder. Are these really required? Why are we committing them?

File: .yarn/install-state.gz
Comment (mkesavan13) [reply .yarn/install-state.gz:1]: I agree. This is just an optimisation file. Will remove it.

**Comment by mkesavan13** [comment]
> I'm unable to load all the files for review.
> Additionally, there seems to be a lot of demo bundles inside the docs/ folder. Are these really required? Why are we committing them?

@sreenara - These are available in the master branch as well and is part of the chore commit for release. So, I'd like to keep it as it is for this PR and come back to inspect it while we modernize the WebexMeetings widget

**Review by sreenara — APPROVED** [review]
Let's fix the changelog in a later PR if needed.

File: packages/@webex/widgets/CONTRIBUTING.md
[the remark this answers is not in the document]
Comment (mkesavan13) [reply packages/@webex/widgets/CONTRIBUTING.md:1]: I've updated the `contributing.md` as well

Just one part is left in the Linking components locally section, `Install peer dependencies`

I'll have to test this to update. Will do so and update it and then merge the PR
